crates/view/src/storage.rs

```diff
@@ -1714,8 +1715,29 @@ impl Storage {
             pool.get()?
                 .prepare_cached(&q)?
                 .query_and_then([], |row| {
-                    let position_id: Vec<u8> = row.get("position_id")?;
-                    Ok(position::Id(position_id.as_slice().try_into()?))
+                    let position_id_bytes: Vec<u8> = row.get("position_id")?;
+                    let id = position::Id(position_id_bytes.as_slice().try_into()?);
+
+                    let position_state_str: String = row.get("position_state")?;
+                    let trading_pair_str: String = row.get("trading_pair")?;
+
+                    // TODO: DB currently does not save full position. Hence, default backups are used below.
+                    //       We should save the full position in the db to be able to provide the full correct data.
```

Comment:
I understand if relying on defaults may not be an acceptable change and we will want to update the database (hard to tell how much work this is though).

Reply:
The tricky part here is that there's currently nothing that comes across in sync that would tell you when a position's reserves changed.

The two approaches that come to mind are:

1. adding a list of all touched positions during a block to the compact block

this seems like it runs the danger of bloating the compact block with data that is not relevant to nearly every client syncing, so I think it should be excluded from consideration

2. adding an additional step to the sync to call against APIs to fetch current position data

This is a little bad because we can't ensure that the block height of the position data is perfectly consistent with the sync block height, but I think it might be fine.

However, this represents a significant shift in view server data, which to my understanding, is currently built entirely on compact block data. Do we want to head down the path of enriching it with data from other RPCs?

Reply:
> leave the table as-is and have the web client fetch the complete, up-to-date position data from the existing RPCs based on the position ID retrieved from the view server

Late to the party, but that feels like a very appropriate solution to the problem at hand. We already have [penumbra.core.component.dex.v1.QueryService.LiquidityPositionById](https://buf.build/penumbra-zone/penumbra/docs/main:penumbra.core.component.dex.v1#penumbra.core.component.dex.v1.QueryService.LiquidityPositionById) which exists specifically for this use case: accept position IDs, return full positions. In OP, @grod220 did say 

> To support limit orders in the web, we need the ability for the view service to return [position](https://buf.build/penumbra-zone/penumbra/docs/main:penumbra.core.component.dex.v1#penumbra.core.component.dex.v1.Position) info to a client.

but I lack context on why that might be a hard requirement. To my eye, we already have "the ability for the view service to return position info to a client."

Reply:
Helpful discussion! 

@zbuc ah, I see what you mean. For point 2, it is not clear to me at what frequency or signal would trigger re-fetching a position's data 🤔 (or how to do this in a privacy-preserving way).

@conorsch thanks for sharing that. I did run into this rpc method. However, before I realized the reserves limitation, I thought it would be simple to just serve the position data that is already in the database. I was gravitating toward updating the `OwnedPositionIds` view service RPC as this does not reveal my positions to the node. However, if there isn't a reasonable way to have up-to-date positions in the db without specifically querying the node with position ID's, it sounds like a fresh query is all the same.

Reply:
@grod220 the naive solution would be to fetch all the positions during `handle_block` alongside the check for new app parameters. And yes this does introduce privacy concerns as you are essentially linking your IP address to all those position IDs. The same problem seems to exist if we do it in the web too. Splitting them apart over time doesn't really fix the issue either.

For privacy's sake, either adding the positions to the compact block or doing something like having the client fetch additional randomly selected positions as "noise" to reduce linkability seems good, but there still seems to be danger of linkability over time if the client doesn't consistently include the same set of noise positions each time it queries.

Reply:
Given there doesn't seem like a practical privacy-preserving solution, it sounds like the roundtrip `OwnedPositionIds -> LiquidityPositionById` query is the way to go for now. Closing this PR. 

Thanks everyone! 🙏

Reply:
@hdevalence tagging you for posterity re: the accepted LP linkability risk for the present moment. IMO it is worth investigating what a better privacy preserving design here would look like.